### ISSUE TITLE
PhantomJS fails to redirect when redirect URL has special character

### ISSUE THREAD

**Issue by engstrand** [ariya/phantomjs#13663]
I am using phantomJS 2.0 on windows together with PHP.

When I try page.open with this url:
"http://www.webhallen.com/se-sv/film/dvd/148610-galpagos?atcl=start:viewed"

The server responds with a redirect to:
http://www.webhallen.com/se-sv/film/dvd/148610-gal%E1pagos
(In firefox the HTTP header location says: "/se-sv/film/dvd/148610-galápagos")

However, phantomJS is messing upp the special character "á" and the redirect URL becomes:
"http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos"

When phantomJS tries to load this URL the server will respond with the same redirect URL again (The server is setup to respond with a redirect to "http://www.webhallen.com/se-sv/film/dvd/148610-gal%E1pagos" for all URLs that starts with "http://www.webhallen.com/se-sv/film/dvd/148610").

The scripts is now stuck in an infinite redirect loop.

```javascript

var args = require('system').args;

var res = new Object();
res.content = "";
res.error = "";

var page = require('webpage').create();


/* timeout to exit script after 10 seconds */
window.setTimeout(function() { res.error = res.error + "TIME TO STOP! "; console.log(JSON.stringify(res)); phantom.exit();  } , 10000);




page.onResourceRequested = function(requestData, request) {
    res.error = res.error + "REQUEST: " + requestData.url + "\r\n";
};



page.onResourceReceived = function(response) {
    res.error = res.error + "RESPONSE: " + response.url + "\r\n";
};



page.open(args[1], {  encoding: "utf8" }, function(status) {
    if (status !== 'success') {
        res.error = res.error + 'No success...' + status;
    } else {
        res.error = res.error + 'SUCCESS!';
        
        res.content = page.content;
        console.log(JSON.stringify(res));
        phantom.exit();
        
    }
});

```

Output from script:

REQUEST: http://www.webhallen.com/se-sv/film/dvd/148610-galpagos?atcl=start:viewed
RESPONSE: http://www.webhallen.com/se-sv/film/dvd/148610-galpagos?atcl=start:viewed
REQUEST: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos
RESPONSE: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos
REQUEST: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos
RESPONSE: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos
REQUEST: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos
RESPONSE: http://www.webhallen.com/se-sv/film/dvd/148610-gal%EF%BF%BDpagos

**Comment by zackw** [comment]
I believe this is another manifestation of bug #12216.  If so, that is fixed by pull request #12233, which will be included in the upcoming 2.0.1 release.  If you are able to build and test the development sources and confirm whether your problem is solved, that would be helpful.  I am going to close this bug report now, but if your problem is _not_ solved, feel free to reopen it.

**Comment by sp90** [comment]
@zackw i dunno if im doing it wrong but im getting this old error by doing this:

```javascript
var system = require('system');
var page = require('webpage').create();
var args = system.args;


page.onResourceRequested = function(requestData, request) {
    console.log("requestData.url: ", requestData.url);
}; 

page.onResourceReceived = function(response) {
    //console.log("response.url: ", response.url);
};


if (args.length === 1) {
	console.log('Try to pass some arguments when invoking this script!');
	phantom.exit(1);
} else {
	var url = args[1];


	console.log("url: ", url);

	page.open(url, {
		encoding: 'utf8'
	}, function(status) {
		// Check for page load success
		if (status !== "success") {
			console.log("Unable to access network");
			phantom.exit(1);
		} else {


			console.log("url: ", url);
			//phantom.exit(0);
		}
	});
}
```

The url i put in is:
`https://bolighed.dk/håndværker`

The url i get out is:
`requestData.url:  https://bolighed.dk/h%EF%BF%BDndv%EF%BF%BDrker`

It seems like its bugging with **Æ, Ø** or **Å**
